starters/gatsby-wordpress-starter/src/components/paginator.jsx

```diff
@@ -0,0 +1,232 @@
+import React from 'react'
+import { useState, useEffect } from 'react'
+import { Link } from 'gatsby'
+
+/**
+ *
+ * @param props.data The data to be paginated
+ * @param props.itemsPerPage The number of items to display on each page
+ * @param props.breakpoints Breakpoints has 3 properties: start, end, and add. Set to {} for no breakpoint.
+ * start: where to start the breakpoint
+ * end: where to end the breakpoint
+ * add: how manu buttons to add when the breakpoint is clicked
+ * ***
+ * note: (`add` * x) + `start` = `end` where x is a number of clicks it takes to fill in all of the buttons
+ * For example: If there are 25 buttons and the start = 5 and end = 25, then add should be 5 or 10.
+ * ***
+ * @param props.routing If true, shallow routing will be enabled. Check the examples/pagination route to see it in action
+ * @param props.Component React Component that takes in currentItems as props and maps over them.
+ * currentItems is a subset of data, so any component that works for data will work here.
+ * @param location holds flag variable to track breakpoint state
+ * @param navRoute route of the page using the paginator
+ * @returns Component with data rendered by the passed in Component and page buttons
+ */
+const Paginator = ({
+	data,
+	itemsPerPage,
+	breakpoints,
+	Component,
+	routing,
+	location,
+	navRoute,
+}) => {
+	// Track if breakpoint buttons have been opened
+	const openStart = location.state?.breakOpen
+		? breakpoints?.start + breakpoints?.add
+		: undefined
+	const [breakStart, setBreakStart] = useState(
+		openStart || breakpoints?.start || null,
+	)
+	const [breakpointsOpen, setBreakpointsOpen] = useState(
+		location.state?.breakOpen || false,
+	)
+	// This value will be the button to start with after separator
+	const breakEnd = breakpoints?.end || null
+	// how many buttons to add when the separator is clicked
+	const breakAdd = breakpoints?.add || null
+
+	const urlParams = window.location.pathname.split('/')
+
+	const [currentPageQuery, setCurrentPageQuery] = useState(
+		routing ? Number(urlParams[urlParams.length - 1]) : 1,
```

Comment:
`Number(urlParams[urlParams.length - 1])` This feels very fragile, can we use a regex or some other way to get the page number from the url?